### PR TITLE
Add missing resources 2021 10 01

### PR DESCRIPTION
**Pull request by anat-gilenson**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [X] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    - [X] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [X] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/common/IncidentTypes.json
Comment (mentat9) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/common/IncidentTypes.json:11]: >enum [](http://example.com/codeflow?start=7&length=4)

You removed the Critical enum value from this definition in previous API version. This is a breaking change and requires signoff from breaking change board. I've added the BreakingChangeReviewRequired label to your PR. Look in the comments for instructions on how to get the review.

**Comment by mentat9** [comment]
This PR was almost entirely a big refactoring (aside from one enum I commented). This made ARM review very difficult. Most of the changes don't need an API version change or ARM review. since refactoring the files doesn't modify the REST APIs. In future it will be **very** helpful to you and the ARM reviewers to do this kind of change in two separate PRs:
1. Refactor your files. When you submit, fill out the submission form indicating that the REST layer is not changing, and your change is only to move files and contents around. This does not require an API version change.
2. After your PR merges, submit another PR that makes the REST changes in a new API version (in this case, just changing the one enum). Of course, be sure to follow the proper construction of the PR by making the first commit be all of the files from the previous API version (just refactored) copied to the new API version folder.

Following this approach will allow our automation to work properly and make the ARM review job trivial which help minimize the time it takes to complete your PR.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/examples/alertRuleTemplates/GetAlertRuleTemplateById.json
Comment (laithhisham) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/examples/alertRuleTemplates/GetAlertRuleTemplateById.json:26]: both the eventGroupingSettings and entityMappings should be part of the template properties schema

**Review by laithhisham — CHANGES_REQUESTED** [review]

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/common/IncidentTypes.json
Comment (anat-gilenson) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/common/IncidentTypes.json:11]: Thank you for the review @mentat9.
`Critical` isn't a real Incident severity and having it in the enum was a mistake already corrected and approved in the 2020-01-01 version ([https://github.com/Azure/azure-rest-api-specs/pull/15761/files](https://github.com/Azure/azure-rest-api-specs/pull/15761/files)). This branch was generated with the OpenAPI hub before the correction so it includes the mistake.

I applied a request to the breaking change board to review this PR.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/examples/alertRuleTemplates/GetAlertRuleTemplateById.json
Comment (anat-gilenson) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-10-01/examples/alertRuleTemplates/GetAlertRuleTemplateById.json:26]: They are missing in the definitions. Will you please add them and correct the example in a separate PR?

**Comment by anat-gilenson** [comment]
> This PR was almost entirely a big refactoring (aside from one enum I commented). This made ARM review very difficult. Most of the changes don't need an API version change or ARM review. since refactoring the files doesn't modify the REST APIs. In future it will be **very** helpful to you and the ARM reviewers to do this kind of change in two separate PRs:
> 
> 1. Refactor your files. When you submit, fill out the submission form indicating that the REST layer is not changing, and your change is only to move files and contents around. This does not require an API version change.
> 2. After your PR merges, submit another PR that makes the REST changes in a new API version (in this case, just changing the one enum). Of course, be sure to follow the proper construction of the PR by making the first commit be all of the files from the previous API version (just refactored) copied to the new API version folder.
> 
> Following this approach will allow our automation to work properly and make the ARM review job trivial which help minimize the time it takes to complete your PR.

Thank you. Next time I'll do the refactoring in a separate PR.

Please note that much of the changes that might seem as refactoring included change of models (different parameter pattern values etc..). I used the common-type definitions and parameters instead of defining our own. I couldn't make these changes in a previous version as it would be considered breaking change.

**Comment by mentat9** [comment]
> > This PR was almost entirely a big refactoring (aside from one enum I commented). This made ARM review very difficult. Most of the changes don't need an API version change or ARM review. since refactoring the files doesn't modify the REST APIs. In future it will be **very** helpful to you and the ARM reviewers to do this kind of change in two separate PRs:
> > 
> > 1. Refactor your files. When you submit, fill out the submission form indicating that the REST layer is not changing, and your change is only to move files and contents around. This does not require an API version change.
> > 2. After your PR merges, submit another PR that makes the REST changes in a new API version (in this case, just changing the one enum). Of course, be sure to follow the proper construction of the PR by making the first commit be all of the files from the previous API version (just refactored) copied to the new API version folder.
> > 
> > Following this approach will allow our automation to work properly and make the ARM review job trivial which help minimize the time it takes to complete your PR.
> 
> @mentat9 Sorry for making it difficult to review. Next time I'll do the refactoring in a separate PR.
> 
> Please note that much of the changes that might seem as refactoring included change of models (different parameter pattern values etc..). I used the common-type definitions and parameters instead of defining our own. I couldn't make these changes in a previous version as it would be considered breaking change.

Makes sense, thanks.

File: specification/securityinsights/resource-manager/readme.md
Comment (qiaozha) [inline specification/securityinsights/resource-manager/readme.md:29]: why do we make this change ?

Comment (anat-gilenson) [reply specification/securityinsights/resource-manager/readme.md:29]: I didn't downgrade, I just didn't change the way it's in main. OpenAPIHub automatically changes the default tag but I don't want to change it. I upgraded the basic tag it in another preview version dev branch and I don't want any conflicts when I try to merge this branch to main.
Readme in [main](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/securityinsights/resource-manager/readme.md):
image

Readme in [2021-09-01-preview dev branch](https://github.com/Azure/azure-rest-api-specs/tree/Microsoft.SecurityInsights-2021-09-01-preview-rebased/specification/securityinsights/resource-manager/readme.md): 
image

Comment (qiaozha) [reply specification/securityinsights/resource-manager/readme.md:29]: Got it, so you just chenge it back to be consistent with the main branch.

**Review by qiaozha — APPROVED** [review]

**Review by laithhisham — APPROVED** [review]

**Pull request merged**
